subprojects/packagefiles/nanobind/meson.build

```diff
@@ -0,0 +1,87 @@
+project(
+    'nanobind',
+    'cpp',
+    version: '2.0.0',
+    meson_version: '>=0.53.0',
+    license: 'BSD-3-Clause',
+)
+
+py_mod = import('python')
+py = py_mod.find_installation()
+py_dep = py.dependency(embed: true)
+
+robin_map_dep = dependency('robin-map')
+
+incdir = include_directories('include')
+
+# Arguments are derived from nanobind_build_library function in CMake configuration
+dep_compile_args = []
+dep_link_args = []
+if get_option('python.allow_limited_api')
+  py_version = py.language_version()
+
+  if py_version.version_compare('>=3.12')
+    add_project_arguments('-DPy_LIMITED_API=0x030C0000', language: 'cpp')
+  endif
+endif
+
+if get_option('buildtype') == 'shared'
+  # nanobind_link_options uses linker response files for either PyPy or
+  # CPython at this step - how do we support in Meson?
+  add_project_arguments('-DNB_BUILD', language: 'cpp')
+  dep_compile_args += ['-DNB_SHARED']
+
+  if host_machine.system() == 'darwin'
+    add_project_link_arguments('-Wl,-dead_strip', '-Wl,x', '-Wl,-S', language: 'cpp')
+  elif host_machine.system() != 'windows'
+    add_project_link_arguments('-Wl,-s', language: 'cpp')
+  endif
+elif host_machine.system() != 'windows' and host_machine.system() != 'darwin'
+  dep_compile_args += ['-ffunction-sections', '-fdata-sections']
+  dep_link_args += ['-Wl,--gc-sections']
+endif
+
+# Here the CMake configuration sets PIC - does Meson handle this internally?
+
+compiler = meson.get_compiler('cpp')
+if compiler.get_id() == 'msvc'
+  dep_compile_args += ['-D_CRT_SECURE_NO_WARNINGS']
+else
+  # Discussion in WrapDB PR #1556 as to whether or not this is necessary
+  # add_project_arguments('-fno-strict-aliasing', language: 'cpp')
+endif
+
+# Here the CMake configuration links against Python::SABIModule on Windows
+# if building an abi3 target - does Meson handle this internally?
+
+if get_option('buildtype') == 'release'
+  add_project_arguments('-DNDEBUG', '-DNB_COMPACT_ASSERTIONS', language: 'cpp')
+endif
```

Comment:
`get_option('b_ndebug')` has 3 values:  true, false, if-release

Probably, projects should simply specify if-release for their own use. There's no need to manually pass -DNDEBUG IMO.

Opinions vary, apparently. :shrug: